Code/src/main/java/nl/utwente/group10/ui/gestures/InputAnchorHandler.java

```diff
@@ -0,0 +1,129 @@
+package nl.utwente.group10.ui.gestures;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import nl.utwente.group10.ui.CustomUIPane;
+import nl.utwente.group10.ui.components.Connection;
+import nl.utwente.group10.ui.components.ConnectionAnchor;
+import nl.utwente.group10.ui.components.ConnectionLine;
+import nl.utwente.group10.ui.components.InputAnchor;
+import nl.utwente.group10.ui.components.OutputAnchor;
+import javafx.event.EventHandler;
+import javafx.geometry.Point2D;
+import javafx.scene.input.InputEvent;
+import javafx.scene.input.MouseDragEvent;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.input.TouchEvent;
+
+public class InputAnchorHandler implements EventHandler<InputEvent> {
+
+	public static final Integer MOUSE_ID = 0;
+	private CustomUIPane cpane;
+	private InputAnchor inputAnchor;
+	private Map<Integer, ConnectionLine> lines;
+
+	public InputAnchorHandler(InputAnchor inputAnchor, CustomUIPane cpane) {
+		this.cpane = cpane;
+		this.inputAnchor = inputAnchor;
+		lines = new HashMap<Integer, ConnectionLine>();
+		inputAnchor.addEventFilter(MouseEvent.ANY, this);
+		inputAnchor.addEventFilter(TouchEvent.ANY, this);
+	}
+
+	@Override
+	public void handle(InputEvent event) {
```

Comment:
Ik vind deze if-boom niet echt mooi, als je dat zou opsplitsen in verschillende event handlers kun je misschien ook meteen het punt hierboven aanpakken.